File: src/commands/projects.js
'use strict';

const { getAuth, getVaultUrl } = require('../config');

async function list(args) {
  const auth = getAuth();
  if (!auth || !auth.token) {
    console.error('Not logged in. Run: vde login');
    process.exit(1);
  }

  const vaultUrl = getVaultUrl(args);

  const resp = await fetch(`${vaultUrl}/api/v1/dashboard/projects`, {
    headers: { 'Authorization': `Bearer ${auth.token}` },
  });

  if (resp.status === 401) {
    console.error('Session expired. Run: vde login');
    process.exit(1);
  }

  if (!resp.ok) {
    console.error(`Error: ${resp.status} ${await resp.text()}`);
    process.exit(1);
  }

  const { projects } = await resp.json();

  if (!projects || projects.length === 0) {
    console.log('No projects found. Create one with: vde init');
    return;
  }

  console.log(`\nProjects (${projects.length}):\n`);
  for (const p of projects) {
    const envs = (p.environments || []).map(e => e.name).join(', ');
    console.log(`  ${p.name}`);
    console.log(`    ID: ${p.id}`);
    if (envs) console.log(`    Environments: ${envs}`);
    console.log();
  }
}

module.exports = { list };

File: clients/python/vaultdotenv/commands/projects.py
"""CLI projects command: list all projects."""
from __future__ import annotations

import sys

import httpx

from vaultdotenv.config import get_auth, get_vault_url


def list_projects(args: list[str]) -> None:
    """List all projects linked to your account."""
    auth = get_auth()
    if not auth or not auth.get("token"):
        print("Not logged in. Run: vde login")
        sys.exit(1)

    vault_url = get_vault_url(args)

    resp = httpx.get(
        f"{vault_url}/api/v1/dashboard/projects",
        headers={"Authorization": f"Bearer {auth['token']}"},
    )

    if resp.status_code == 401:
        print("Session expired. Run: vde login")
        sys.exit(1)

    if not resp.is_success:
        print(f"Error: {resp.text}")
        sys.exit(1)

    projects = resp.json().get("projects", [])

    if not projects:
        print("No projects found. Create one with: vde init")
        return

    print(f"\nProjects ({len(projects)}):\n")
    for p in projects:
        envs = ", ".join(e["name"] for e in p.get("environments", []))
        print(f"  {p['name']}")
        print(f"    ID: {p['id']}")
        if envs:
            print(f"    Environments: {envs}")
        print()
